dotnet/Cucumber.Messages.Specs/ProtocolVersionTest.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Cucumber.Messages.Specs
+{
+    public class ProtocolVersionTest
+    {
+        [Fact]
+        public void ProtocolVersionIsCorrect()
+        {
+            Assert.Matches(@"\d+\.\d+\.\d+(-SNAPSHOT)?", Cucumber.Messages.ProtocolVersion.Version);
```

Comment:
Does .Net use snapshot versions?

Reply:
It is not a community standard, no. Version numbering conventions are team-specific. Most teams do use some kind of suffix to designate pre release builds. 

I blindly copied what the Java implementation was doing. Sorry. I should have investigated what the prior .Net implementation had done and carried that forward. 

I'll revise as needed.

Reply:
The prior version also used a suffix called SNAPSHOT; or at least that's what I can make of it (sorry, not well versed with github workflows). Open to changing that to whatever makes sense. I will ask Gaspar about it on Monday.

Reply:
No worries. I think that's a bit of copy paste from Gherkin. 

You can leave the `.csproj` file as is for now. I'll make a separate item to clean it up. It also means we need to change our release scripts that match against `VersionNumber` instead of `Version`.

https://github.com/cucumber/polyglot-release/blob/75e8af2c3398433d7a01862a863d038aa7d22dcb/polyglot-release#L70

Reply:
Generalized that regex to look for an optional dot separated set of alphanumeric identifiers per the Semver standard as commonly used in .net apps.